agent/agents/mysql/slowlog/parser/parser.go

```diff
@@ -167,90 +168,121 @@ func (p *SlowLogParser) Run() {
+		p.parseAdmin(line)
+
+	default:
+		p.parseMetrics(line)
+	}
+}
+
+func skipPrefix(line string) string {
+	line = line[2:]
+	return line
+}
+
+func (p *SlowLogParser) parseTime(line string) {
+	p.logf("time")
+	m := timeRe.FindStringSubmatch(line)
+	if len(m) == 2 {
+		p.event.Ts, _ = time.ParseInLocation("060102 15:04:05", m[1], p.opts.DefaultLocation)
+	} else {
+		m = timeNewRe.FindStringSubmatch(line)
 		if len(m) == 2 {
-			p.event.Ts, _ = time.ParseInLocation("060102 15:04:05", m[1], p.opts.DefaultLocation)
+			p.event.Ts, _ = time.ParseInLocation(time.RFC3339Nano, m[1], p.opts.DefaultLocation)
 		} else {
-			m = timeNewRe.FindStringSubmatch(line)
-			if len(m) == 2 {
-				p.event.Ts, _ = time.ParseInLocation(time.RFC3339Nano, m[1], p.opts.DefaultLocation)
-			} else {
-				return
-			}
-		}
-		if userRe.MatchString(line) {
-			p.logf("user (bad format)")
-			m := userRe.FindStringSubmatch(line)
-			p.event.User = m[1]
-			p.event.Host = m[2]
-		}
-
-	case strings.HasPrefix(line, "# User"):
-		p.logf("user")
-		m := userRe.FindStringSubmatch(line)
-		if len(m) < 3 {
 			return
 		}
+	}
+	if userRe.MatchString(line) {
+		p.logf("user (bad format)")
+		m := userRe.FindStringSubmatch(line)
 		p.event.User = m[1]
 		p.event.Host = m[2]
+	}
+}
 
-	case strings.HasPrefix(line, "# admin"):
-		p.parseAdmin(line)
+func (p *SlowLogParser) parseUser(line string) {
+	p.logf("user")
+	m := userRe.FindStringSubmatch(line)
+	if len(m) < 3 {
+		return
+	}
+	p.event.User = m[1]
+	p.event.Host = m[2]
+}
 
-	default:
-		p.logf("metrics")
-		submatch := schema.FindStringSubmatch(line)
-		if len(submatch) == 2 {
-			p.event.Db = submatch[1]
-		}
+func (p *SlowLogParser) parseMetrics(line string) {
+	p.logf("metrics")
+	submatch := schema.FindStringSubmatch(line)
+	if len(submatch) == 2 {
+		p.event.Db = submatch[1]
+	}
+
+	if !strings.Contains(line, ":") {
+		return
+	}
 
-		m := metricsRe.FindAllStringSubmatch(line, -1)
-		for _, smv := range m {
-			switch {
-			// [String, Metric, Value], e.g. ["Query_time: 2", "Query_time", "2"]
-			case strings.HasSuffix(smv[1], "_time") || strings.HasSuffix(smv[1], "_wait"):
-				// microsecond value
-				val, _ := strconv.ParseFloat(smv[2], 64)
-				p.event.TimeMetrics[smv[1]] = val
-
-			case smv[2] == "Yes" || smv[2] == "No":
-				// boolean value
-				if smv[2] == "Yes" {
-					p.event.BoolMetrics[smv[1]] = true
-				} else {
-					p.event.BoolMetrics[smv[1]] = false
-				}
-
-			case smv[1] == "Schema":
-				p.event.Db = smv[2]
-
-			case smv[1] == "Log_slow_rate_type":
-				p.event.RateType = smv[2]
-
-			case smv[1] == "Log_slow_rate_limit":
-				val, _ := strconv.ParseUint(smv[2], 10, 64)
-				p.event.RateLimit = uint(val)
-
-			default:
-				// integer value
-				val, _ := strconv.ParseUint(smv[2], 10, 64)
-				p.event.NumberMetrics[smv[1]] = val
+	line = strings.Replace(line, ": ", ":", -1)
+	for _, kv := range strings.Split(line, " ") {
+		if len(kv) == 0 {
+			continue
+		}
+		kv2 := strings.Split(kv, ":")
```

Comment:
kv, kv2... it is hard to understand logic with this names. can you please refactor it a bit?